src/java/org/apache/cassandra/config/DatabaseDescriptor.java

```diff
@@ -1925,15 +1942,16 @@ static long getDataFileDirectoriesMinTotalSpaceInGB(Multiset<FileStore> fileStor
         }).min().orElse(0L) * fileStores.size();
     }
 
-    public static String getCommitLogLocation()
+    public static File getCommitLogLocation()
     {
-        return conf.commitlog_directory;
+        return commitlogDirectory;
     }
 
     @VisibleForTesting
-    public static void setCommitLogLocation(String value)
+    public static void setCommitLogLocation(File value)
     {
-        conf.commitlog_directory = value;
+        commitlogDirectory = value;
+        conf.commitlog_directory = value.path();
```

Comment:
In other methods conf does not seem to be touched - maybe tests can update conf separately if it is needed?